### ISSUE TITLE
Add a lock for button

### ISSUE THREAD

**Issue by SeByDocKy** [esphome/feature-requests#1761]
Would be fantastic to have a lock propeties for button object.

For example, a button with such option can simulate a double press action within a time period. On the first press, the button becomes unlocked, then the second press do the action. If the second press is not here within let's say 2s, the button returns locked.

**Additional context**

**Comment by bkaufx** [comment]
I don't really understand this as a lock property, because how do you unlock a button by pressing the button that is locked?  Do you just want the button to require a double press instead of single press to activate?  You could implement this pretty easily by creating a script that has just a 2s (let's say) delay.  When the button is pressed, you check to see if this script is running.  If the script is not running, you just run the script.  If the script is running, then you do whatever action you want to do with the button.

**Comment by bkaufx** [comment]
If the script is running, after you do whatever it is you want to do with the button, you could also stop the script so that the next press would be another unlock press even if it was still within the same 2s.

**Comment by mmakaay** [comment]
Yes, the described functionality is highly specific it seems, and there's no actual locking going on. A lock for me would be that the button doesn't respond to anything (which can also be solved easily within the YAML).

A script would be feasible for solving that, but as described here, to me it sounds like simply a slow double click requirement. That can be solved with [on_double_click](https://esphome.io/components/binary_sensor/index.html#on-double-click) withi some appropriate timing parameters I'd say.

```yaml
binary_sensor:
  - platform: gpio
    # ...
    on_double_click:
      min_length: 50ms
      max_length: 2000ms
      then:
        - do-the-thing:
```

**Comment by mmakaay** [comment]
@SeByDocKy Do the provided directions for a solution help you, or is there something that we missed?
If these solve your requirement, then we can close this request.
